### PR TITLE
PR: Adapt package for renaming (fork)

### LINKED ISSUE #1: Package rename
We need to do these tasks in order to rename the package

- [x] Change package name in setup.py to `spyder-pyls`.
- [x] Change Readme to describe how to develop this package with Spyder.
- [x] Remove versioneer.py, add release instructions and add a tuple with the current version to `_version.py`.
- [x] Set initial version to 0.90 to differentiate from `python-language-server`.
- [x] Set CIs
- [x] Add changelog for version 0.90.
- [x] Remove old branches (except `develop`)

### PR DESCRIPTION
**Pull request by goanpeca**
Fixes #1

### CODE REVIEW

**Comment by goanpeca** [comment]
@ccordoba12 just to double check what did you mean by updating the README for spyder dev?

Remove all the VSCode related stuff and replace by spyder?

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# History of changes
+
+## Version 0.9 (2019/03/30)
```
Comment (ccordoba12) [inline CHANGELOG.md]: I want the first release number for our fork to be `0.90`, not `0.9`

File: pyls/_version.py [hunk 2]
```diff
@@ -1,520 +1,9 @@
...
-            "dirty": None,
-            "error": "unable to compute version", "date": None}
+# -*- coding: utf-8 -*-
+# ----------------------------------------------------------------------------
+# Copyright (c) Spyder Project Contributors
+#
+# Licensed under the terms of the MIT License
+# ----------------------------------------------------------------------------
+
+VERSION_INFO = (0, 9, 0, "dev0")
```
Comment (ccordoba12) [inline pyls/_version.py]: Same here, please change this to be `0.90.0`

File: setup.py [hunk 3]
```diff
@@ -1,25 +1,43 @@
...
+# Standard library imports
+import ast
+import os
+
+# Third party imports
 from setuptools import find_packages, setup
-import versioneer
 
+HERE = os.path.abspath(os.path.dirname(__file__))
 README = open('README.rst', 'r').read()
```
Comment (ccordoba12) [inline setup.py]: Let's write our Readme in markdown instead of rest.

Comment (ccordoba12) [reply setup.py]: For that, you also need to add another line to the setup function:

https://github.com/spyder-ide/qtpy/pull/186/files#diff-2eeaed663bd0d25b7e608891384b7298R37

File: setup.py [hunk 4]
```diff
@@ -1,25 +1,43 @@
...
 
     description='Python Language Server for the Language Server Protocol',
 
     long_description=README,
 
     # The project's main homepage.
-    url='https://github.com/palantir/python-language-server',
+    url='https://github.com/spyder/spyder-pyls',
 
     author='Palantir Technologies, Inc.',
```
Comment (ccordoba12) [inline setup.py]: This should be now `Palantir Technologies, Inc. and the Spyder development team`.

**Comment by ccordoba12** [comment]
> Remove all the VSCode related stuff and replace by spyder?

Yes, for now please:

1. Remove all VSCode screenshots and text (this means all images under the `resources` folder can be removed here too).
2. Add two Spyder screenshots: one showing code completion and another one showing warnings/errors with the mouse hovering in one of them to be able to see its message. 
3. Write instructions about how to develop this package with Spyder:
    1. Install it with `pip install -e .`
    1. Start Spyder in debug mode (`python bootstrap.py --debug`)
    2. Check logs for both client and server in `~/.spyder-py3/lsp-logs`

File: setup.py [hunk 5]
```diff
@@ -69,9 +87,9 @@
     # pip to create the appropriate form of executable for the target platform.
     entry_points={
         'console_scripts': [
-            'pyls = pyls.__main__:main',
+            'spyder-pyls = pyls.__main__:main',
```
Comment (ccordoba12) [inline setup.py]: Let's revert this change. That way Spyder will remain compatible with PyLS and all its plugins.

File: setup.py [hunk 6]
```diff
@@ -69,9 +87,9 @@
...
         ],
-        'pyls': [
+        'spyder_pyls': [
```
Comment (ccordoba12) [inline setup.py]: Let's also revert this change because it could break PyLs plugins.

File: pyls/__init__.py [hunk 7]
```diff
@@ -2,12 +2,10 @@
...
 import pluggy
-from ._version import get_versions
+from ._version import __version__
 
+__version__ = __version__
 install_aliases()
-__version__ = get_versions()['version']
-del get_versions
-
 PYLS = 'pyls'
```
Comment (ccordoba12) [inline pyls/__init__.py:9]: Please add the following constant below this line

    SPYDER_PYLS = True

**Review by ccordoba12 — COMMENTED** [review]
@goanpeca, I left some more comments to make Spyder compatible with both Palantir's and our own package.

**Comment by goanpeca** [comment]
Made the changes @ccordoba12

**Review by ccordoba12 — COMMENTED** [review]
The Readme is looking really good!! Thanks @goanpeca!

Please readd the line you removed in PR #12, then I'd say this is ready.

**Review by ccordoba12 — APPROVED** [review]
Thanks @goanpeca!
